src/libs/actions/Session/index.js

```diff
@@ -316,7 +317,14 @@ function setPassword(password, validateCode, accountID) {
                 return;
             }
 
-            Onyx.merge(ONYXKEYS.ACCOUNT, {error: Localize.translateLocal('setPasswordPage.accountNotValidated')});
+            const login = lodashGet(response, 'data.email', null);
+            Onyx.merge(ONYXKEYS.ACCOUNT, {
+                validated: true, accountExists: true, validateCodeExpired: true, error: Localize.translateLocal('setPasswordPage.accountNotValidated'),
```

Comment:
Return `true` for `validated`? The `validateCode` is expired so they are not validated...?

Reply:
This block gets called when validation code is expired and the account is validated. This flow comes from `ValidateEmail`, only if the `Account Validated`.

https://github.com/Expensify/App/blob/01b7b60387dcdfb8286a25d598ad3a5a62af8004/src/libs/actions/Session/index.js#L413

Hence, I've updated `validated: true`

Reply:
That explanation doesn't make sense to me. We are setting an error here that says:

"We were unable to validate your account."

And the account is being set as `validated: true`.

Reply:
Not sure if I explained it right. Let me give it another short. In `ResendValidtionForm` we have the following checks:

1. `isNewAccount`
2. `isOldUnvalidatedAccount` -> This gives the error about "Account exists but not validated"
3. `validateCodeExpired` - Handled by this PR, which shows `validationCodeFailedMessage` message
4. The default `weSentYouMagicSignInLink` magic link text.

Now the two instances of  `accountNotValidated` translation key were added during the `n6-hold` merge, which I haven't changed. If you see the `setPassword` function call is called only after we get the error as `Email is already validated`. Hence I am setting `validated: true`. 


In short if I ask, if I remove the `error: Localize.translateLocal('setPasswordPage.accountNotValidated')` from the `SetPassword` function and just retain it in the line 433.

```
Onyx.merge(ONYXKEYS.ACCOUNT, {
                validated: true, accountExists: true, validateCodeExpired: true,
}); // remove error from here.
```

**Retain this:**
```
Onyx.merge(ONYXKEYS.SESSION, {error: 'setPasswordPage.accountNotValidated'});
```


Does the above change make sense?

Reply:
I've made the above change and tested it. It is working fine.

Reply:
>Does the above change make sense?

It makes sense as a quick solution. However, based on the full conversation in this PR so far - it isn't the one we want and will introduce inconsistency with how we handle errors.

Please try to ask clarifying questions before proposing alternatives, because I'm not sure if we are seeing eye to eye on the problem. I will try to be as specific as possible so we can move on from this PR...

The issue is with setting **`validated: true`**.

It seems the only reason we need to set this here is so that the correct error will show here:

https://github.com/Expensify/App/blob/361f097a8d1cb7077774b2f951278ca00f694cae/src/pages/signin/ResendValidationForm.js#L90-L105

The `isOldUnvalidatedAccount` variable needs to be `false` and I suspect that's only reason we are setting it to `true`. But **it is _not_ the correct state for the account**. We must change the logic that shows the errors not the state of the account.